src/services/review/RepoCheckerService.ts

```diff
@@ -345,62 +356,59 @@ export default class RepoCheckerService {
               const anchorTags = dom.window.document.querySelectorAll("a")
               forEach(anchorTags, (tag) => {
                 const href = tag.getAttribute("href")
-                const text = tag.textContent || ""
-                if (!href) {
+                const text = tag.textContent
+
+                if (!text) {
+                  // while rendered in the dom, unlikely to be
+                  // noticed by end citizen.
+                  return
+                }
+                if (!tag.hasAttribute("href")) {
                   const error: RepoError = {
                     type: RepoErrorTypes.BROKEN_LINK,
                     linkToAsset: "",
                     viewablePageInCms: this.getPathInCms(fileName, repo),
-                    viewablePageInStaging: path.join(filePermalink),
+                    viewablePageInStaging: path.join(normalisedPermalink),
                     linkedText: text,
                   }
                   errors.push(error)
-                } else {
-                  const decodedHref = this.mediaPathDecoder(href)
-                  if (
-                    !this.isExternalLinkOrPageRef(decodedHref) &&
-                    !setOfAllMediaAndPagesPath.has(decodedHref) &&
-                    // Amplify supports adding of trailing slash
-                    !setOfAllMediaAndPagesPath.has(`${decodedHref}/`)
-                  ) {
-                    const error: RepoError = {
-                      type: RepoErrorTypes.BROKEN_LINK,
-                      linkToAsset: href,
-
-                      viewablePageInCms: this.getPathInCms(fileName, repo),
-                      viewablePageInStaging: path.join(filePermalink),
-                      linkedText: text,
-                    }
-                    errors.push(error)
+                }
+                if (!href) {
+                  // could be intentional linking to self
+                  // eg. <a href>Back to top</a>
+                  return
+                }
```

Comment:
i saw that the users are using empty hrefs - just wanted to check if [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#onclick_events) is what they intended? 

if so, we should report an error (not necessarily a broken link error, but something else certainly) so that they can use a button or something more suited for the job.

wdyt?

Reply:
hey thanks for bringing this up! abit of early feedback that we have received is that the errors are not true errors, and this has led to agency not fixing the issues.  
currently the product direction for this feature is that we want to make it as easy and as least overwhelming as possible to the users, which is why we are deprio any non-obvious errors to users first, and think we can revisit this again maybe thr design (maybe nice to haves vs actual errors)

Reply:
yeap, i think what's important here is _who_ does the determination of errors in ambiguous cases like these. 

as you've rightfully pointed out, some agencies **don't** think that this is an error and rightfully so! but in the case of agencies that just forgot to put a link in, these may actually be errors. 

in that case, rather than a blanket judgement on our end to deem these as non-errors, perhaps what would be more useful here is a categorisation of the severities of the broken links that the checker flags out? 

eg: if your `img` or `a` tag is missing a link entirely, it's flagged as informational, but if those tags have a link pointing to a missing item (eg: `href="some_missing_item"`), we flag it as error. 

that seems to fit what the agency wants, as well as still helping out potential broken links use case

Reply:
while I am aligned with you that users should be aware of the less severe errors in the long run, I might have to push back on this for this pr. looking at the datadog, we saw instances of repo owners looking at the broken links and then proceeding to approve and merge changes to production 

as of right now, the users of this feature in production have yet to fix the higher priority bugs. the discussion of how to tackle low prior bugs shd occur after we see evidence of agency users tackling high prior bugs. we can then have a discussion with design regarding how to display the not so severe errors, what are your thoughts?

Reply:
> the discussion of how to tackle low prior bugs shd occur after we see evidence of agency users tackling high prior bugs

yeap, that's fine but the approach taken here is to declassify them as bugs, which seems to run counter to what you're saying ._.

we can just leave these checks in first then instead of pre-emptively removing them?

Reply:
discussed offline, filing ticket for this!

Reply:
copy pasting from slack discussion: 
1. we shouldn't enable user laziness - regardless of whether it is a bug or not, the user should always be the one making the judgement rather than us determining that it isn't and aiding them to avoid that judgement
2. this shouldn't be a permanent measure - just a temporary one, with clear hypothesis of what this should encourage/discourage and follow-ups from there. 

i'm ok with this **as a temporary measure** - and we need to be clear about this. i've filed two tickets tracking this and we should have a clear yes/no decision following on